pkgs/applications/office/tradingview/update.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl gnused gnugrep jq
+
+set -eu -o pipefail
+
+dirname=$(dirname "$0" | xargs realpath)
+nixpkgs=$(realpath "${dirname}/../../../..")
+attr=tradingview
+nix_file="$dirname/default.nix"
+
+channel="${1:-stable}" # stable/candidate/edge
+
+# create bash array from snap info
+snap_info=($(
+  curl -s -H 'X-Ubuntu-Series: 16' \
+    "https://api.snapcraft.io/api/v1/snaps/details/tradingview?channel=$channel" \
+  | jq --raw-output \
+    '.revision,.download_sha512,.version,.last_updated'
+))
+
+# "revision" is the actual version identifier on snapcraft, the "version" is
+# just for human consumption. Revision is just an integer that gets increased
+# by one every (stable or unstable) release.
+upstream_revision="${snap_info[0]}"
+sha512="${snap_info[1]}"
+upstream_version="${snap_info[2]}"
+last_updated="${snap_info[3]}"
+sri_hash=$(nix hash to-sri --type sha512 $sha512)
```

Comment:
Why are we using sha512? Standard is sha256.

Also why are we converting it in the first place? If upstream has a sha512 that works then we should use it without base64 encoding. This makes things a lot easier to double check.